### PR TITLE
chore: plumb the target close reason when test fails

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: packages/playwright-core/src/client/browser.ts [hunk 1]
```diff
@@ -130,12 +130,12 @@ export class Browser extends ChannelOwner<channels.BrowserChannel> implements ap
     return buffer;
   }
 
-  async close(): Promise<void> {
+  async close(options: { reason?: string } = {}): Promise<void> {
     try {
       if (this._shouldCloseConnectionOnClose)
         this._connection.close();
```
Comment (dgozman) [inline packages/playwright-core/src/client/browser.ts:136]: Might as well plumb it here too.

File: packages/playwright-core/src/server/fetch.ts [hunk 2]
```diff
@@ -91,6 +91,7 @@ export abstract class APIRequestContext extends SdkObject {
   readonly fetchLog: Map<string, string[]> = new Map();
   protected static allInstances: Set<APIRequestContext> = new Set();
   readonly _activeProgressControllers = new Set<ProgressController>();
+  _closeReason: string | undefined;
```
Comment (dgozman) [inline packages/playwright-core/src/server/fetch.ts:94]: Plumb this one in a follow up?

File: packages/playwright/src/index.ts [hunk 3]
```diff
@@ -374,7 +375,7 @@ const playwrightFixtures: Fixtures<TestFixtures, WorkerFixtures> = ({
     const context = await (browser as any)._newContextForReuse(defaultContextOptions);
     (context as any)[kIsReusedContext] = true;
     await use(context);
-    await (browser as any)._stopPendingOperations('Test ended');
+    await (browser as any)._stopPendingOperations('Test ended.');
```
Comment (dgozman) [inline packages/playwright/src/index.ts:378]: This one could also be `Test timeout of ...`

**Review by dgozman — APPROVED** [review]

File: packages/playwright-core/src/client/browser.ts (hunk 1 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/client/browser.ts:136]: I'll handle client side separately.

File: packages/playwright-core/src/server/fetch.ts (hunk 2 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/server/fetch.ts:94]: Can do.

**Pull request merged**